### PR TITLE
Throw error when entity attribute and embedded attribute have name clashes

### PR DESCRIPTION
**Pull request by KyleAure**
- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

Handle and test the following scenarios: 
```java
public class Residence {
    public Occupant occupant;
    public boolean isOccupied;
    public static class Occupant {
        public String firstName;
        public String lastName;
    }
}
public class Apartment extends Residence {
    public long aptId;
    public int quartersWidth; // Warning - shadows the embedded class need to use delimited attribute name in query/order/sort
    public int quarters_width; // Failure - shadows the embedded class, user would be incapable of query/order/sort on embedded field
    public boolean occupant_firstName; // Failure - shadows the super-embedded class, user would be incapable of query/order/sort on super-embedded field
    public Bedroom quarters;
}
public class Bedroom {
    public int width;
    public int length;
}

```

### CODE REVIEW

File: dev/io.openliberty.data.internal_fat_jpa/test-applications/DataJPATestApp/src/test/jakarta/data/jpa/web/DataJPATestServlet.java [hunk 1]
```diff
@@ -2928,6 +2932,132 @@ public void testOneToOne() {
         drivers.deleteByFullNameEndsWith(" TestOneToOne");
     }
 
+    @Test
+    public void testPersistentFieldNamesAndDelimiters() {
```
Comment (njr-11) [inline dev/io.openliberty.data.internal_fat_jpa/test-applications/DataJPATestApp/src/test/jakarta/data/jpa/web/DataJPATestServlet.java:2941]: Add a Javadoc comment to this test with a sentence stating what it covers. That information, along with the test name, will eventually be copied into the Feature Test Summary to get our test approval.
```suggestion
    /**
     * Tests entity attribute names from embeddables and MappedSuperclass that
     * can have delimiters. Includes tests for name collisions with attributes from an
     * embeddable or superinteface.
     */
    @Test
    public void testPersistentFieldNamesAndDelimiters() {
```

File: dev/io.openliberty.data.internal_fat/test-applications/DataTestApp/src/test/jakarta/data/web/DataTestServlet.java [hunk 2]
```diff
@@ -3693,6 +3719,132 @@ public void testParameterAnnotationTakesPrecedenceOverMethodPrefix() {
         assertEquals(19L, nine.numberId);
     }
 
+    @Test
```
Comment (njr-11) [inline dev/io.openliberty.data.internal_fat/test-applications/DataTestApp/src/test/jakarta/data/web/DataTestServlet.java:3727]: Need a comment that can go into the feature test summary. Something like:
```suggestion
    /**
     * Tests entity attribute names from embeddables and MappedSuperclass that
     * can have delimiters. Includes tests for name collisions with attributes from an
     * embeddable or superinteface. This test uses unannotated entities.
     */
    @Test
```

File: dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/EntityManagerBuilder.java [hunk 3]
```diff
@@ -114,148 +119,175 @@ protected void collectEntityInfo(Set<Class<?>> entityTypes) throws Exception {
...
+                                idClassAttributeAccessors = getIdClassAccessors(idType, idClassAttributes);
+                            }
                         }
                     }
+                } catch (MappingException e) {
+                    collectionException = e;
+                } catch (Exception e) {
+                    if (trace & tc.isDebugEnabled())
+                        Tr.debug(tc, "Unexpected exception caught while collecting entity information. The entity information will complete exceptionally", e);
+                    collectionException = e;
```
Comment (njr-11) [inline dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/EntityManagerBuilder.java:290]: This is a note for us to optimize this path and cover a related scenario later.  Don't worry about addressing it now.  It isn't related to your changes, but I happened to notice it while reviewing.
```suggestion
                    collectionException = e;
                    // TODO compute the userEntityClass earlier and completeExceptionally here.
                    // Then move the EntityInfo creation and successful completion to before the catch.
                    // That will also cover the possibility of EntityInfo constructor failing,
                    // which can happen when it doesn't validate.
```

**Review by njr-11 — DISMISSED** [review]
This looks great. I added a few doc comments, one of which is a note to me for the future to cover an additional error path.

**Comment by KyleAure** [comment]
#libby - check headers
#build - personal build

**Review by njr-11 — APPROVED** [review]

**Pull request merged**